mlflow/pipelines/pipeline.py

```diff
@@ -45,6 +45,9 @@ def __init__(self, pipeline_root_path: str, profile: str) -> None:
         self._profile = profile
         self._name = get_pipeline_name(pipeline_root_path)
         self._steps = self._resolve_pipeline_steps()
+        self._template = get_pipeline_config(self._pipeline_root_path, self._profile).get(
+            "template"
```

Comment:
We might want to rename `template` to something else, but it doesn't have to be done in this PR.

Reply:
Will add a TODO